cpp/models/seir_ide/seir_ide.h

```diff
@@ -0,0 +1,128 @@
+/* 
+* Copyright (C) 2020-2021 German Aerospace Center (DLR-SC)
+*
+* Authors: Lena Ploetzke
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef SEIR_IDE_H
+#define SEIR_IDE_H
+
+#include "memilio/math/eigen.h"
+#include "memilio/utils/time_series.h"
+#include "memilio/epidemiology/contact_matrix.h"
+
+#include <vector>
+
+namespace mio
+{
+class IdeModel
+{
+public:
+    /**
+    * Constructor of IDEModel
+    * @param init TimeSeries containing the initial (time, quantity of susceptible) values
+    *   the initial time steps should be equidistant with distance dt_init; 
+    *   we need values from time -(m_k-1)*dt until 0 for our calculation (with m_k=std::ceil((m_infectious_time+m_latency_time)/dt)) 
+    *   -> first initial time point should be earlier than -(m_k-1)*dt
+    * @param dt_init size of the time step used for numerical integration.
+    * @param N_init Population of the considered region. 
+    */
+    IdeModel(TimeSeries<double>&& init, double dt_init, int N_init);
+
+    /**
+    * Change m_latency_time, default value: 3.3
+    * @param latency The value of the new latency time
+    */
+    void set_latency_time(double latency);
+
+    /**
+    * Change m_infectious_time, default value: 8.2
+    * @param infectious The value of the new infectious time
+    */
+    void set_infectious_time(double infectious);
+
+    /**
+    * Get Reference to used ContactMatrix (effective Contactfrequency is used = quantity of Contacts * probability of infection in case of contact)
```

Comment:
```suggestion
    * Returns the ContactMatrix of the IDE model. (effective Contactfrequency is used = quantity of Contacts * probability of infection in case of contact)
```
Do not write if reference is returned or not. This should be clear from the "ContactMatrix&". Avoid parentheses. Do you have a better suggestion than mine above?